packages/lesswrong/server/styleGeneration.tsx

```diff
@@ -12,13 +12,47 @@ import crypto from 'crypto'; //nodejs core library
 import draftjsStyles from '../themes/globalStyles/draftjsStyles';
 import miscStyles from '../themes/globalStyles/miscStyles';
 import { isValidSerializedThemeOptions, ThemeOptions, getForumType } from '../themes/themeNames';
-import type { ForumTypeString } from '../lib/instanceSettings';
+import { ForumTypeString } from '../lib/instanceSettings';
 import { getForumTheme } from '../themes/forumTheme';
 import { usedMuiStyles } from './usedMuiStyles';
-import { minify } from 'csso';
 import { requestedCssVarsToString } from '../themes/cssVars';
+import { isGithubActions } from '../lib/executionEnvironment';
+import { exec } from "child_process";
+import { Stream } from 'stream';
+import CleanCSS from "clean-css";
 
-const generateMergedStylesheet = (themeOptions: ThemeOptions): Buffer => {
+/**
+ * Minify the CSS bundle,
+ * Ordinarily, we do this by invoking cleancss as a separate cli process to
+ * maximise the amount of parallelism we can get during server startup.
+ * When we're running in Github CI, we instead do the conversion in-process as
+ * we can't run multiple processes simultaneously.
+ */
+const minifyCss = (css: string): Promise<string> | string =>
+  isGithubActions ? minifyCssSync(css) : minifyCssAsync(css);
+
+const minifyCssSync = (css: string): string => new CleanCSS().minify(css).styles;
+
+const minifyCssAsync = async (css: string): Promise<string> => {
```

Comment:
This one makes me want to definitely check staging before deploying. ([Like we should](https://www.notion.so/centreforeffectivealtruism/Dev-Process-0174eb89b31d4879a8e8843312b9606f).)